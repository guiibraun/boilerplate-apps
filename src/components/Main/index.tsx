import Image from 'next/image'
import logo from '@/assets/logo.svg'

export default function Main({
  title = 'React Avançado',
  description = 'Crie aplicações reais com NextJS, Strapi, GraphQL e mais!'
}) {
  return (
    <main className="bg-indigo-950 w-full p-8 text-white min-h-screen flex-col gap-8 flex justify-center items-center">
      <Image src="/logo.svg" alt="" width={250} height={250} />
      <h1 className=" text-6xl font-bold ">{title}</h1>
      <p>{description}</p>
    </main>
  )
}
