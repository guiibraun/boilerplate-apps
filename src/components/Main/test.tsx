import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    //renderiza o component
    const { container } = render(<Main />)
    //busca o elemento
    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()
    //verifica a existencia
    expect(container.firstChild).toMatchSnapshot()
  })
})
