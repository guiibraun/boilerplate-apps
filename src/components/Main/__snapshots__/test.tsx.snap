// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="bg-indigo-950 w-full p-8 text-white min-h-screen flex-col gap-8 flex justify-center items-center"
>
  <img
    alt=""
    data-nimg="1"
    decoding="async"
    height="250"
    loading="lazy"
    src="/logo.svg"
    style="color: transparent;"
    width="250"
  />
  <h1
    class=" text-6xl font-bold "
  >
    React Avançado
  </h1>
  <p>
    Crie aplicações reais com NextJS, Strapi, GraphQL e mais!
  </p>
</main>
`;
